test/app-luatest/http_client_test.lua

```diff
@@ -52,101 +50,129 @@ local function start_server(test, sock_family, sock_addr)
         end
         fiber.sleep(0.01)
     end
-    test:is(r.status, 200, "connection is ok")
-    if r.status ~= 200 then
-        os.exit(1)
-    end
+    t.assert_equals(r.status, 200, "connection is ok")
     return server, url, opts
 end
 
-local function stop_server(test, server)
-    test:diag("stopping HTTP server")
-    server:close()
-end
+g.before_all(function(cg)
+    local sock_family = cg.params.sock_family
+    local sock_addr
+    if sock_family == 'AF_INET' then
+        local s = socketlib('AF_INET', 'SOCK_STREAM', 0)
+        s:bind('127.0.0.1', 0)
+        sock_addr = string.format("%s:%d", s:name().host, s:name().port)
+        s:close()
+    else
+        assert(sock_family == 'AF_UNIX')
+        local path = os.tmpname()
+        os.remove(path)
+        local status = pcall(client.get, 'http://localhost/',
+                             {unix_socket = path})
+        t.skip_if(not status, "not supported")
```

Comment:
This is quite strange condition. If we'll broke unix sockets support in the http client, we'll not spot it in the test.